packages/backend/src/webhook/service.ts

```diff
@@ -160,3 +172,18 @@ export function generateWebhookSignature(
 
   return `t=${timestamp}, v${version}=${digest}`
 }
+
+async function getWebhookEventsPage(
+  deps: ServiceDependencies,
+  options?: GetPageOptions
+): Promise<WebhookEvent[]> {
+  const { filter, pagination } = options ?? {}
+  const { type } = filter ?? {}
+  const query = WebhookEvent.query(deps.knex)
+
+  if (type) {
+    query.where({ type })
+  }
```

Comment:
Still thinking through where and how to handle applying the filters to the query.

Either we manually apply each one in the service like I'm doing here, which seems reasonable for very simple filters like this, or we create some utility function that applies them. I think this utility will be necessary if we do something like:

```gql
input WebhookEventFilter {
  type: {eq: String, contains: String, startsWith: String etc.}
  withdrawalAmount: {gt: UInt64, lt: UInt64, eq: Uint64 etc.}
}
```

Seconds, is where to apply them. If doing the utility that can handle any given filter, I wonder about passing the filter into `baseModel.getPage` and call the utility internally. This would save calling the function in every service and ensure it's always applied correctly (before pagination). Additionally, if we need to encode the filter in the cursor (TBD) then we'll need to pass the filter in anyways.

Reply:
I think this is the right place for them for now :+1:. If we start adding more filters to different models, we could see a pattern, and how we could abstract it away